### DISCUSSION TITLE
Memory requirements to Build

### DISCUSSION THREAD

**Discussion by pakona** [firecmsco/firecms#180] [Q&A] [answered]
Greetings,

I recently discovered FireCMS, and I'm excited to try it out on various projects.
I'm stuck on the QuickStart guide though (https://firecms.co/docs/quickstart).

When I try to build the example in that page (`yarn build`), I get the error below

```
yarn run v1.22.17
$ react-scripts --max-old-space-size=4092 build
Creating an optimized production build...

<--- Last few GCs --->

[57730:0x1048d7000]   146379 ms: Scavenge (reduce) 4062.6 (4084.3) -> 4062.0 (4084.3) MB, 197.3 / 0.0 ms  (average mu = 0.389, current mu = 0.381) allocation failure 
[57730:0x1048d7000]   146387 ms: Scavenge (reduce) 4062.8 (4081.3) -> 4062.2 (4082.5) MB, 6.5 / 0.0 ms  (average mu = 0.389, current mu = 0.381) allocation failure 
[57730:0x1048d7000]   146463 ms: Scavenge (reduce) 4063.2 (4081.7) -> 4062.6 (4082.7) MB, 74.2 / 0.0 ms  (average mu = 0.389, current mu = 0.381) allocation failure 


<--- JS stacktrace --->

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 0x101306bb5 node::Abort() (.cold.1) [/Users/macmini/.nvm/versions/node/v14.17.5/bin/node]
 2: 0x1000ad6c9 node::Abort() [/Users/macmini/.nvm/versions/node/v14.17.5/bin/node]
 3: 0x1000ad82f node::OnFatalError(char const*, char const*) [/Users/macmini/.nvm/versions/node/v14.17.5/bin/node]
 4: 0x1001f1b97 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [/Users/macmini/.nvm/versions/node/v14.17.5/bin/node]
 5: 0x1001f1b33 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/Users/macmini/.nvm/versions/node/v14.17.5/bin/node]
 6: 0x10039f695 v8::internal::Heap::FatalProcessOutOfMemory(char const*) [/Users/macmini/.nvm/versions/node/v14.17.5/bin/node]
 7: 0x1003a113a v8::internal::Heap::RecomputeLimits(v8::internal::GarbageCollector) [/Users/macmini/.nvm/versions/node/v14.17.5/bin/node]
 8: 0x10039c865 v8::internal::Heap::PerformGarbageCollection(v8::internal::GarbageCollector, v8::GCCallbackFlags) [/Users/macmini/.nvm/versions/node/v14.17.5/bin/node]
 9: 0x10039a190 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/Users/macmini/.nvm/versions/node/v14.17.5/bin/node]
10: 0x1003a88ba v8::internal::Heap::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/Users/macmini/.nvm/versions/node/v14.17.5/bin/node]
11: 0x1003a8941 v8::internal::Heap::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/Users/macmini/.nvm/versions/node/v14.17.5/bin/node]
12: 0x100376712 v8::internal::Factory::NewFillerObject(int, bool, v8::internal::AllocationType, v8::internal::AllocationOrigin) [/Users/macmini/.nvm/versions/node/v14.17.5/bin/node]
13: 0x1006f6930 v8::internal::Runtime_AllocateInOldGeneration(int, unsigned long*, v8::internal::Isolate*) [/Users/macmini/.nvm/versions/node/v14.17.5/bin/node]
14: 0x100a7c3f9 Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_NoBuiltinExit [/Users/macmini/.nvm/versions/node/v14.17.5/bin/node]
15: 0x30d342b57af9 
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I've tried to change the build command from:

`react-scripts build` to `react-scripts --max-old-space-size=4092 build`

But no luck.
What are the memory requirements for building FireCMS projects?

Thanks!

**Comment by fgatti675** [comment]
Hi @pakona 
That's weird... I have tried running one of my projects with `react-scripts --max-old-space-size=2048 build` and it has worked fine, also using node 14. 
I do get the same error with a heap size of 1024 though.
Do you have any more dependencies?

**Comment by pakona** [comment]
Hi @fgatti675,

I'm running with the same dependencies indicated in the quick start guide. The only modification I made is the one for the heap size.
Here is my package.json:

```
{
  "name": "app_cms",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@camberi/firecms": "^1.0.0-beta8",
    "@emotion/react": "^11.7.1",
    "@emotion/styled": "^11.6.0",
    "@mui/icons-material": "^5.1.0",
    "@mui/lab": "^5.0.0-alpha.55",
    "@mui/material": "^5.1.0",
    "@mui/styles": "^5.1.0",
    "@testing-library/jest-dom": "^5.16.1",
    "@testing-library/react": "^12.1.2",
    "@testing-library/user-event": "^13.5.0",
    "@types/jest": "^27.0.3",
    "@types/node": "^16.11.13",
    "@types/react": "^17.0.37",
    "@types/react-dom": "^17.0.11",
    "firebase": "^9.0.0",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-router": "^6.0.0",
    "react-router-dom": "^6.0.0",
    "react-scripts": "5.0.0",
    "typescript": "^4.5.4",
    "web-vitals": "^2.1.2"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts --max-old-space-size=4092 build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}
```

App.tsx is verbatim what is in this link: https://firecms.co/docs/quickstart/

Note: I'm on OSX, in case that makes a difference.

**Comment by fgatti675** [comment]
This is very strange...
I have tried a fresh install with no issues.
I post my `yarn.lock` file in case you want to copy exactly my resolved dependencies and see if it works
(just remove the .txt extension and replace yours and run `yarn`)
[yarn.lock.txt](https://github.com/Camberi/firecms/files/7728960/yarn.lock.txt)

**Reply by pakona** [reply]
Okay. This was all my fault. 

tldr; I was missing the peer dependencies. 

I was running npm version 6.14.14. [That version doesn't install peer dependencies](https://stackoverflow.com/questions/46053414/npm-warn-requires-a-peer-of-but-none-is-installed-you-must-install-peer),
and I totally ignored the warnings regarding missing peer dependencies, after running `npm install`.

I tried these two fixes and the both worked:

- Manually installed all peer dependencies
- Upgraded to the latest version of npm and reran `npm install`.

Thanks for the support @fgatti675!
I'm now on my way to create some awesome dashboards :)

**Reply by fgatti675** [reply]
Great thanks! Let me know if you need help!

**Comment by pakona** [comment, marked as answer]
Marking this as fixed!
